### DISCUSSION TITLE
Possible to get `role_connections` without user's authenticating via OAuth2?

### DISCUSSION THREAD

**Discussion by ReenigneArcher** [Pycord-Development/pycord#2667] [Q&A] [answered]
I am working on some updates to my discord bot that allow for better integration with GitHub. https://github.com/LizardByte/support-bot/pull/368

Currently there are a couple of options.

1. User gets `github` role when they approve the "linked role" (https://support.discord.com/hc/en-us/articles/10388356626711-Connections-Linked-Roles-Admins#h_01GK28EMFYFKM4DT7MTFJZC765). However, this option is limited because I can't (or don't know how to) detect the user's GitHub username/id.
![image](https://github.com/user-attachments/assets/a504e2b4-dc9f-4192-a6ff-854d17e0b529)
![image](https://github.com/user-attachments/assets/bbe0aa71-1529-4338-bb8a-8e3043b01dce)

2. User uses a slash command to authenticate via Discord's OAauth2 option, and as long as they have connected GitHub to their account I can get the username/id.

3. User uses a slash command to authenticate via GitHub's OAuth2 option, and I can link the Discord user to their GitHub username/id.

Option 2 and 3 are not ideal because I need to provide a callback URL, and would like to avoid exposing the bot in this manner (if possible).

Is there any way to get the GitHub username when using option 1?

**Comment by Lulalaby** [comment, marked as answer]
Linked roles is a special feature.
From the bots perspective, it allows u to build ur own linked role for stuff that has not an official connection within discord.

To get the built-in data from the connection, in this case github, you **have to** use oauth.

It's personal information. Otherwise every bot could just scrape data and create identifiable profiles about users. Which would go against the terms of service / GDPR.

So tl;Dr, you have to use oauth

**Reply by ReenigneArcher** [reply]
Got it, thanks. I was hoping it would be possible using the linked roles feature since the user not only has to connect the account, but also explicitly grant permission to each server to get the linked role.

**Reply by Lulalaby** [reply]
yeah i get that.
from developers perspective, it sadly only grants the server itself the permission
but from users perspective, it's safer this way
